src/components/VmActions/Action.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import Button from './Button'
+import { getOnlyComponentProps } from '../utils'
+
+class Action extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = { showModal: false }
+    this.handleOpen = this.handleOpen.bind(this)
+    this.handleClose = this.handleClose.bind(this)
+  }
+
+  handleOpen () {
+    this.setState({ showModal: true })
+  }
+
+  handleClose () {
+    this.setState({ showModal: false })
+  }
+
+  render () {
+    const { children, confirmation } = this.props
+    let trigger = children
+    let confirmationDialog = confirmation || null
+    if (confirmation) {
+      trigger = React.cloneElement(trigger, { onClick: this.handleOpen })
+      confirmationDialog = React.cloneElement(confirmationDialog, { show: this.state.showModal, onClose: this.handleClose })
+    }
+    return <React.Fragment>
+      {trigger}
+      {confirmationDialog}
+    </React.Fragment>
+  }
+}
+
+Action.propTypes = {
+  children: PropTypes.node.isRequired,
+  confirmation: PropTypes.node,
+}
+
+export default Action
+
+const ActionButtonWraper = (props) => {
+  const btnProps = getOnlyComponentProps(props, Button)
```

Comment:
````javascript
  import { excludeKeys } from 'patternfly-react'

  const btnProps = excludeKeys(props, [ 'confirmation' ])
````
That would be sufficient to keep those properties from going to your button and then we would not need the extra function in our `helpers.js`.

[All of the patternfly-react helpers here](https://github.com/patternfly/patternfly-react/blob/patternfly-react%402.4.1/packages/core/src/common/helpers.js).  They get exported to the default namespace, so they can just be straight imported.